### PR TITLE
prow/pr_status: clear cookie sessions of invalid access token

### PR DESCRIPTION
**Pull request by qhuynh96**

### CODE REVIEW

File: prow/prstatus/prstatus.go [hunk 1]
```diff
@@ -168,20 +169,44 @@ func (da *DashboardAgent) HandlePrStatus(queryHandler PullRequestQueryHandler) h
...
+			// to validate whether the access token is consumable or not. If not, invalidate the
+			// session.
 			grc := pullRequestRestfulClient{ghclient.NewClient(token.AccessToken, false)}
-			login, err := getUserLogin(session, grc)
+			var err error
+			login, err = getUserLogin(session, grc)
 			if err != nil {
-				serverError("Error with getting user login", err)
-				return
+				if strings.Contains(err.Error(), "401 Unauthorized") {
```
Comment (BenTheElder) [inline prow/prstatus/prstatus.go]: hrm this feels slightly brittle.

File: prow/prstatus/prstatus.go [hunk 2]
```diff
@@ -168,20 +169,44 @@ func (da *DashboardAgent) HandlePrStatus(queryHandler PullRequestQueryHandler) h
...
 		if ok && token.Valid() {
-			data.Login = true
-			ghc := github.NewClient(token.AccessToken, githubEndpoint)
-			// Get user login
+			// If access token exist, get user login using the access token. This is a chance
+			// to validate whether the access token is consumable or not. If not, invalidate the
+			// session.
 			grc := pullRequestRestfulClient{ghclient.NewClient(token.AccessToken, false)}
-			login, err := getUserLogin(session, grc)
+			var err error
```
Comment (rmmh) [inline prow/prstatus/prstatus.go]: why is this line necessary?

File: prow/prstatus/prstatus.go [hunk 3]
```diff
@@ -168,20 +169,44 @@ func (da *DashboardAgent) HandlePrStatus(queryHandler PullRequestQueryHandler) h
...
 		}
+
+		login := ""
 		if ok && token.Valid() {
-			data.Login = true
-			ghc := github.NewClient(token.AccessToken, githubEndpoint)
-			// Get user login
+			// If access token exist, get user login using the access token. This is a chance
+			// to validate whether the access token is consumable or not. If not, invalidate the
+			// session.
```
Comment (cjwagner) [inline prow/prstatus/prstatus.go]: The `getUserLogin` function doesn't necessarily get the user login using the access token. It might just use the stored session value in which case validation is skipped?

File: prow/prstatus/prstatus.go [hunk 4]
```diff
@@ -168,20 +169,44 @@ func (da *DashboardAgent) HandlePrStatus(queryHandler PullRequestQueryHandler) h
...
 			grc := pullRequestRestfulClient{ghclient.NewClient(token.AccessToken, false)}
-			login, err := getUserLogin(session, grc)
+			var err error
+			login, err = getUserLogin(session, grc)
 			if err != nil {
-				serverError("Error with getting user login", err)
-				return
+				if strings.Contains(err.Error(), "401 Unauthorized") {
+					// Invalidate access token session
+					session.Options.MaxAge = -1
```
Comment (rmmh) [inline prow/prstatus/prstatus.go]: Would it make more sense to redirect to /github-login here?

Comment (rmmh) [reply prow/prstatus/prstatus.go]: Oh, that's handled in javascript. Ok.

File: prow/prstatus/prstatus.go (hunk 2 above)
Comment (stevekuznetsov) [reply prow/prstatus/prstatus.go]: Otherwise the `:=` below makes it less clear what is being shadowed?

Comment (rmmh) [reply prow/prstatus/prstatus.go]: Ah, missed the scope of login.

File: prow/prstatus/prstatus.go (hunk 1 above)
Comment (rmmh) [reply prow/prstatus/prstatus.go]: Maybe match for 401 instead?

File: prow/prstatus/prstatus.go (hunk 3 above)
Comment (qhuynh96) [reply prow/prstatus/prstatus.go]: I changed it to request for user login everytime
